packages/components/src/number-control/index.js

```diff
@@ -93,58 +93,47 @@ export function NumberControl(
 				nextValue = subtract( nextValue, incrementalValue );
 			}
 
-			nextValue = roundClamp( nextValue, min, max, incrementalValue );
-
-			state.value = nextValue;
+			state.value = constrainValue( nextValue );
 		}
 
 		/**
 		 * Handles drag to update events
 		 */
 		if ( type === inputControlActionTypes.DRAG && isDragEnabled ) {
-			const { delta, shiftKey } = payload;
-			const [ x, y ] = delta;
-			const modifier = shiftKey
+			const modifier = payload.shiftKey
 				? parseFloat( shiftStep ) * baseStep
 				: baseStep;
 
 			let directionModifier;
-			let directionBaseValue;
+			let delta;
 
 			switch ( dragDirection ) {
 				case 'n':
-					directionBaseValue = y;
+					delta = payload.delta[ 1 ];
 					directionModifier = -1;
 					break;
 
 				case 'e':
-					directionBaseValue = x;
+					delta = payload.delta[ 0 ];
 					directionModifier = isRTL() ? -1 : 1;
 					break;
 
 				case 's':
-					directionBaseValue = y;
+					delta = payload.delta[ 1 ];
 					directionModifier = 1;
 					break;
 
 				case 'w':
-					directionBaseValue = x;
+					delta = payload.delta[ 0 ];
```

Comment:
These changes were mostly in interest of formatting for a slight LOC reduction. The destructuring didn't seem helpful and by using `delta` (instead of the verbose `directionBaseValue`) for the value from the axis of interest, it let the new line 133 to fit on a single line.

Reply:
The destructuring of the delta does make it immediately clear which value, `x` or `y`, is being used. Could the line length issue or naming be improved without removing that clarity?

Reply:
Thank you! I brought back the destructure to `x` and `y`.